File: Striver A2Z/Step 13 - Binary Trees [Traversals, Medium and Hard Problems]/Lec 1 - Traversals/binary-tree-inorder-traversal.cpp

// https://leetcode.com/problems/binary-tree-inorder-traversal/

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    void findInorderTraversal(TreeNode* root, vector<int>&ans){
        if(!root) return;
        findInorderTraversal(root->left, ans);
        ans.push_back(root->val);
        findInorderTraversal(root->right, ans);
    }
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> ans;
        if(!root) return ans;
        findInorderTraversal(root, ans);
        return ans;
    }
};

File: Striver A2Z/Step 13 - Binary Trees [Traversals, Medium and Hard Problems]/Lec 1 - Traversals/binary-tree-inorder-traversal.js

// https://leetcode.com/problems/binary-tree-inorder-traversal/

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var solve = (root, inorder) => {
    if(!root) return;
    solve(root.left, inorder);
    inorder.push(root.val);
    solve(root.right, inorder);
    return;
}

var inorderTraversal = function(root) {
    const inorder = [];
    if(!root) return inorder;
    solve(root, inorder);
    return inorder;
};